See upstream PR: https://github.com/scipy/scipy/pull/21069
Fixes:
```
../scipy/_lib/_uarray/small_dynamic_array.h(145,18): error: reference to non-static member function must be called
  145 |     size_ = copy.size;
      |             ~~~~~^~~~
1 error generated.
```
diff --git a/scipy/_lib/_uarray/small_dynamic_array.h b/scipy/_lib/_uarray/small_dynamic_array.h
index b6c46d7c44fa..351b5d8fc653 100644
--- scipy/_lib/_uarray/small_dynamic_array.h.orig
+++ scipy/_lib/_uarray/small_dynamic_array.h
@@ -142,7 +142,7 @@ class SmallDynamicArray {

     clear();

-    size_ = copy.size;
+    size_ = copy.size_;
     try {
       allocate();
     } catch (...) {
